README.md

```diff
@@ -1,22 +1,93 @@
-# PG_Embedding
+# pg_embedding
 
-This ANN extension of Postgres is based
-on [ivf-hnsw](https://github.com/dbaranchuk/ivf-hnsw) implementation of [HNSW](https://www.pinecone.io/learn/hnsw),
+The `pg_embedding` extension enables the use of the Hierarchical Navigable Small World (HNSW) algorithm for vector similarity search in PostgreSQL.
+
+HNSW is a graph-based approach to indexing multi-dimensional data. It constructs a multi-layered graph, where each layer is a subset of the previous one. During a search, the algorithm navigates through the graph from the top layer to the lowest layer to quickly find the nearest neighbor. 
+
+Ths extension is based on [ivf-hnsw](https://github.com/dbaranchuk/ivf-hnsw) implementation of [HNSW](https://www.pinecone.io/learn/hnsw),
 the code for the current state-of-the-art billion-scale nearest neighbor search system<sup>[[1]](#references)</sup>.
 
-## Postgres extension
+## Usage summary
+
+```sql
+CREATE EXTENSION embedding;
+CREATE TABLE documents(id integer PRIMARY KEY, embedding real[]);
+CREATE INDEX ON documents USING hnsw(embedding) WITH (maxelements=1000000, dims=100, m=32);
+SELECT id FROM documents ORDER BY emebedding <-> ARRAY[1.0, 2.0,...] LIMIT 100;
+```
 
-HNSW index is hold in memory (built on demand), and its maximal size is limited by `maxelements` index parameter. Another required parameter is a number of dimensions (if it is not specified in column type).
-Optional parameter `ef` specifies the number of neighbors which are considered during index construction and search (corresponds `efConstruction` and `efSearch` parameters described in the article<sup>[[1]](#references)</sup>).
+## Enable the extension
 
-## Example of usage:
+Enable the `pg_embedding` extension in PostgreSQL by running the following `CREATE EXTENSION` statement.
 
+```sql
+CREATE EXTENSION embedding;
 ```
-create extension embedding;
-create table embeddings(id integer primary key, payload real[]);
-create index on embeddings using hnsw(payload) with (maxelements=1000000, dims=100, m=32);
-select id from embeddings order by payload <-> array[1.0, 2.0,...] limit 100;
+
+## Create a table for vector data
+
+To store vector data, create a table similar to the following:
+
+```sql
+CREATE TABLE documents(id INTEGER, embedding REAL[]);
 ```
 
+This statement generates a table named `documents` with a `embedding` column for storing vector data. (Your table and vector column names may differ.)
+
+## Insert data
+
+To insert vector data:
+
+```sql
+INSERT INTO documents(id, embedding) 
+VALUES 
+(1, '{1.1, 2.2, 3.3,...}'),
+(2, '{4.4, 5.5, 6.6,...}'),
+(3, '{7.7, 8.8, 9.9,...}');
```

Comment:
I don't like ellipsis' because they make SQL invalid, and you can't just copy-paste the code from the example into the SQL Editor and run it.